backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inout/impl/HUPackingMaterialDAO.java

```diff
@@ -89,7 +90,22 @@ public I_M_HU_PackingMaterial retrievePackingMaterialOrNull(@NonNull final Packa
 				.andCollectChildren(I_M_HU_Item.COLUMN_M_HU_ID)
 				.andCollect(I_M_HU_PackingMaterial.COLUMN_M_HU_PackingMaterial_ID, I_M_HU_PackingMaterial.class)
 				.create()
-				// first only used here because we wanna see (read: blow up) if there are cases with 2 packing materials for a single package. I don't think this could happen, but wht if it does?
+				// first only used here because we wanna see (read: blow up) if there are cases with 2 packing materials for a single package. I don't think this could happen, but what if it does?
+				.firstOnly(I_M_HU_PackingMaterial.class);
+	}
+
+	@Nullable
+	@Override
+	public I_M_HU_PackingMaterial retrievePackingMaterialOrNull(@NonNull final HuId huId)
+	{
+		return queryBL
+				.createQueryBuilder(I_M_HU.class)
```

Comment:
createQueryBuilder(I_M_HU.class) shall not be part o tihs DAO

Reply:
what would it take to separate it per DAOs?

Reply:
rewrote to separate DAO concerns.